### ISSUE TITLE
Nuxt: Client-side rendered virtual DOM tree is not matching server-rendered content

### ISSUE THREAD

**Issue by hilyafadhilah** [jarvelov/vue-form-json-schema#82]
Hello, I just installed this package for a working project and I tried following the minimal demo. It worked. However, I got this warning:

`[Vue warn]: The client-side rendered virtual DOM tree is not matching server-rendered content. This is likely caused by incorrect HTML markup, for example nesting block-level elements inside <p>, or missing <tbody>. Bailing hydration and performing full client-side render.`

When I removed the component altogether it's gone. Is it a bug or did I miss some steps?
I'm using Nuxt 2.14 with Vue 2.6. Thanks in advance.

**Comment by jarvelov** [comment]
Hi!

I don't have time to look into it right now but I have a hunch, it might be because of the autogenerated ID which VFJS uses. Try to set `<vue-form-json-schema id="test" />` and see if this fixes it. I'll look into it later, but hopefully setting the id to a static value fixes it.

**Comment by hilyafadhilah** [comment]
Thank you for the fast response 😊

Anyhow, I've found a workaround by using the `<client-only />` tag on top of the component. Now it runs smoothly. But this does mean this plugin doesn't support server side rendering. Maybe you could add that to the docs😄

Also, just wondering, do you plan on supporting it?

**Comment by jarvelov** [comment]
Hi @hilyafadhilah

I definitely think that VFJS should be SSR compatible. I just created a [codesandbox](https://codesandbox.io/s/youthful-perlman-73q2p?file=/pages/index.vue) with `nuxt` and VFJS. However, I did not see that message in the console of the browser or in the terminal, regardless if I ran it in production or in development.

It would be very helpful if you could help me recreate the error you are receiving by updating the codesandbox. I would also like to know what version of `nuxt` you are using.

**Comment by hilyafadhilah** [comment]
Hi again😊

I think I kinda get the problem now. I'm sorry that I entirely forgot about the real and first error that I faced. So, I added your plugin using `nuxt plugins`. I created `~/plugins/vue-form-json-schema.js` which contains these lines:

```
import Vue from 'vue'
import VueFormJsonSchema from 'vue-form-json-schema'

Vue.component('vue-form-json-schema', VueFormJsonSchema)
```

Then, I added this line to my `nuxt.config.js`

```
plugins: [
  // ...other plugins
  '~/plugins/vue-form-json-schema.js',
] ,
```

But when I ran it, it breaks during SSR with this error:

![image](https://user-images.githubusercontent.com/16741377/98273434-d60a9180-1fc4-11eb-99cf-e86851f7487c.png)

So I went and look for answers. Some articles recommended turning off the `ssr` option, like this

```
plugins: [
  // ...other plugins
  { src: '~/plugins/vue-form-json-schema.js', ssr: false }
] ,
```

I thought that made sense, because not all vue plugins are made for SSR right? This is where I messed up. The reason that `Vue warn` shows up was because I'm turning off SSR. And then I completely forgot about that configuration that I made, LOL.

Now, I tried using the same technique as your sandbox, importing the package directly in my page file, and **_it works_**, yay! However when I use `nuxt plugins` it doesn't. I tried replicating the error in the [codesandbox](https://codesandbox.io/s/brave-gauss-ckz53) but it doesn't produce the same `ReferenceError`. It does produce the `Vue warn`, though.

Maybe this is the fault on my part? I don't know why it doesn't work when I attached the component using `nuxt plugins`. I'm just following the [documentation](https://nuxtjs.org/docs/2.x/directory-structure/plugins/#vue-plugins).

What do you think?

**Comment by jarvelov** [comment]
Hi again,

There was a problem with how the import was made in the `plugins/vue-form-json-schema.js`. It is correct that you can not use the default import, i.e. `import VueFormJsonSchema from 'vue-form-json-schema`, and have to import the ESM version directly.

However, when your import is the ESM version, then the `Vue.use` function can not be used, as that file exports the component, not the plugin. Thus the `Vue.component` function should be used instead to register the component. The plugin export is really only used to register the component, so you're not missing out on any features by importing the ESM version directly.

Which means that if we make just a tiny change in the code we can get things up and running. Here's the relevant code

```js
import Vue from "vue";
import VueFormJsonSchema from "vue-form-json-schema/dist/vue-form-json-schema.esm.js";

// Vue.use(VueFormJsonSchema); // Replace this
Vue.component("VueFormJsonSchema", VueFormJsonSchema); // With this
```

Now everything seems to work as expected. See this codesandbox for a working example: https://codesandbox.io/s/pensive-tharp-c4jny

Please check out the codesandbox and let me know if this solves the problems you're facing.

**Comment by hilyafadhilah** [comment]
Hello.

Thank you so much for assistance! Now it's working perfectly. Looking forward to using it😁
